CMakeLists.txt

```diff
@@ -67,6 +69,15 @@ find_package(OpenSSL REQUIRED)
 # Needs to be here before we set further compilation options
 add_subdirectory(3rd-party)
 
+# TODO: remove this when gRPC fixes the compilation with g++ 11.
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  string(REPLACE "." ";" MULTIPASS_GCC_VERSION ${CMAKE_CXX_COMPILER_VERSION})
+  list(GET MULTIPASS_GCC_VERSION 0 MULTIPASS_GCC_VERSION_MAJOR)
```

Comment:
Clever, but why do you check the version? GCC supports `maybe-uninitialized` since at least version 4, so the following should work:

```
if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
  target_compile_options(gRPC INTERFACE "-Wno-error=maybe-uninitialized")
endif()
```

And I think this should be in the 3rd-party CMakeLists.txt instead. There is a target_compile_options for it there already.

Reply:
Yeah, but it's weird it doesn't complain in our code in versions < 11. Maybe the `-Wmaybe-uninitialized` parameter is disabled by some other argument in older versions. Anyway, it doesn't harm to add it for all versions, and it simplifies the code. Changed.

Reply:
Right, it makes an effort to detect such cases, but it offers no guarantees. I suppose they just detect more cases in version 11.